File: src/main/java/training/official/classic_puzzle_easy/HorseRacingDuals.java
package training.official.classic_puzzle_easy;

import java.util.Arrays;
import java.util.Scanner;

class HorseRacingDuals {
	public static void main(String args[]) {
		Scanner in = new Scanner(System.in);
		int N = in.nextInt();

		int[] horsesPower = new int[N];
		for (int i = 0; i < N; i++) {
			horsesPower[i] = in.nextInt();
		}
		Arrays.sort(horsesPower);

		int answer = Integer.MAX_VALUE;
		for (int i = 1; i < horsesPower.length; i++) {
			int currentPowerDiff = horsesPower[i] - horsesPower[i - 1];
			if (currentPowerDiff < answer) {
				answer = currentPowerDiff;
			}
		}

		System.out.println(answer);
	}
}

File: src/main/kotlin/training/official/classic_puzzle_easy/HorseRacingDuals.kt
//package training.official.classic_puzzle_easy

import java.util.*

fun main() {
	val horses = HorseRacingDuals.extractInputData()

	val sortedHorses = horses.sortedBy { it.horsePower }
	var minDiff = Int.MAX_VALUE
	(1 until sortedHorses.size).forEach { i ->
		minDiff = minOf(minDiff, sortedHorses[i].horsePower - sortedHorses[i - 1].horsePower)
	}

	println(minDiff)
}

private class HorseRacingDuals { // only here for codingame
	companion object {
		fun extractInputData(): List<Input> {
			return Scanner(System.`in`).run {
				(0 until nextInt())
					.map { Input(nextInt()) }
					.toList()
			}
		}
	}

	data class Input(val horsePower: Int)
}
